ssg/templates.py

```diff
@@ -30,349 +29,32 @@
+        spec = importlib.util.spec_from_file_location(
+            "preprocess_mod", self.preprocessing_file_path
         )
-    return data
-
-
-@template(["ansible", "bash", "oval", "puppet", "ignition", "kubernetes"])
-def service_disabled(data, lang):
-    if "packagename" not in data:
-        data["packagename"] = data["servicename"]
-    if "daemonname" not in data:
-        data["daemonname"] = data["servicename"]
-    return data
-
-
-@template(["ansible", "bash", "oval", "puppet"])
-def service_enabled(data, lang):
-    if "packagename" not in data:
-        data["packagename"] = data["servicename"]
-    if "daemonname" not in data:
-        data["daemonname"] = data["servicename"]
-    return data
-
-
-@template(["ansible", "bash", "oval", "kubernetes"])
-def sshd_lineinfile(data, lang):
-    missing_parameter_pass = data["missing_parameter_pass"]
-    if missing_parameter_pass == "true":
-        missing_parameter_pass = True
-    elif missing_parameter_pass == "false":
-        missing_parameter_pass = False
-    data["missing_parameter_pass"] = missing_parameter_pass
-    return data
-
-
-@template(["ansible", "bash", "oval"])
-def shell_lineinfile(data, lang):
-    value = data["value"]
-    if value[0] in ("'", '"') and value[0] == value[-1]:
-        msg = (
-            "Value >>{value}<< of shell variable '{varname}' "
-            "has been supplied with quotes, please fix the content - "
-            "shell quoting is handled by the check/remediation code."
-            .format(value=value, varname=data["parameter"]))
-        raise Exception(msg)
-    missing_parameter_pass = data.get("missing_parameter_pass", "false")
-    if missing_parameter_pass == "true":
-        missing_parameter_pass = True
-    elif missing_parameter_pass == "false":
-        missing_parameter_pass = False
-    data["missing_parameter_pass"] = missing_parameter_pass
-    no_quotes = False
-    if data["no_quotes"] == "true":
-        no_quotes = True
-    data["no_quotes"] = no_quotes
-    return data
-
-
-@template(["ansible", "bash", "oval"])
-def timer_enabled(data, lang):
-    if "packagename" not in data:
-        data["packagename"] = data["timername"]
-    return data
-
-
-@template(["oval"])
-def yamlfile_value(data, lang):
-    data["ocp_data"] = data.get("ocp_data", "false") == "true"
-    return data
-
-
-@template(["oval"])
-def argument_value_in_line(data, lang):
-    return data
-
-
-@template(["ansible", "bash", "oval"])
-def zipl_bls_entries_option(data, lang):
-    return data
-
-
-@template(["oval", "kubernetes"])
-def coreos_kernel_option(data, lang):
-    return data
+        preprocess_mod = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(preprocess_mod)
+        parameters = preprocess_mod.preprocess(parameters.copy(), lang)
+        # TODO: Remove this right after the variables in templates are renamed
+        # to lowercase
```

Comment:
I would assume that the intention was to avoid naming conflicting and that the variable came from the input template data. But back then the templating system was different. I don't think it makes much sense nowadays.

Reply:
Yes, it's a remainder from the previous system. We can rename the variables in the templates to lowercase now and then we will be able to remove this part of the code.

Reply:
I see. I propose to remove this obsolete thing in a subsequent PR.

Reply:
Yes